### DISCUSSION TITLE
Dynamicially Change Column Name Defined in Schema

### DISCUSSION THREAD

**Discussion by hantsy** [Kotlin/dataframe#622] [Q&A] [answered]
Given the following data schema defined.

```kotlin 

@DataSchema
data class ExampleDataSchema(

    @ColumnName("NAME")
    val name: String? = null,

    @ColumnName("CREATED AT")
    val createdAt: LocalDateTime? = null
)
```

We try to export CSV content like this:

```kotlin
aList
  .map {
      ExampleDataSchema(
          it.name,
          it.createdAt?.toLocalZone(zoneId)// tolocalzone is an extension used to convert a UTC datetime to local datatime.
      )
  }
  .toDataFrame().toCsv()
```

We want to add the zone info to the column title, and change the Column Name `CREATED AT` in csv to `CREATED AT(Asia/Shanghai)`, is it possible to archive this by modifying the dataframe model?

**Comment by Jolanrensen** [comment, marked as answer]
If I understand you correctly, this requires just two DataFrame calls.

Given some list of data, we first create a DataFrame with your `@DataSchema` class:
```kt
import java.time.ZoneId
val data = listOf(
    ExampleDataSchema("John", java.time.LocalDateTime.now()),
    ExampleDataSchema("Jane", java.time.LocalDateTime.now())
)

val df = data.toDataFrame()
df
```
| NAME | CREATED AT |
| :--- | :--- |
| John | 2024-03-14T11:40:41.609217717 |
| Jane | 2024-03-14T11:40:41.609257253 |

Then, we get the `zoneId` we want to convert to, like 
```kt
val zoneId = ZoneId.systemDefault()
```
You just want to rename the column right? Not update the values? Then it's as simple as:
```kt
val df2 = df
    .rename { `CREATED AT` }.into { "CREATED DATE($zoneId)" }
df2
```
| NAME | CREATED DATE\(Europe/Amsterdam\) |
| :--- | :--- |
| John | 2024-03-14T11:40:41.609217717 |
| Jane | 2024-03-14T11:40:41.609257253 |

and then you can `df2.toCsv()` to export the result. 

Is that what you were looking for?
